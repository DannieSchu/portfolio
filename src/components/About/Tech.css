.Tech {
  margin-left: 6rem;
  height: 100vh;

  section {
    display: flex;
    flex-flow: column wrap;
  }
}
