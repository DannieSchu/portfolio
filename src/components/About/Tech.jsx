import React from 'react';
import TechItem from './TechItem';
import Arrow from '../Nav/Arrow';
import { techArr } from '../../data/techArr';
import styles from './Tech.css';
import typeStyle from '../../styles/type.css';

const Tech = () => {
  const techElements = techArr.map(element => <TechItem key={element.title} {...element} />);

  return (
    <section name="Tech" className={styles.Tech}>
      <h2 className={typeStyle.pageHeading}>Tech Stack</h2>
      <section className={styles.techList}>
        <dl className={styles.definitionList}>
          {techElements}
        </dl>
      </section>
      <Arrow element="Projects" />
    </section>
  );
};

export default Tech;
