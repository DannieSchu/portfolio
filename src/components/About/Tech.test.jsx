import React from 'react';
import { shallow } from 'enzyme';
import Tech from './Tech';

describe('Tech component', () => {
  it('renders Tech', () => {
    const wrapper = shallow(<Tech />);
    expect(wrapper).toMatchSnapshot();
  });
});
