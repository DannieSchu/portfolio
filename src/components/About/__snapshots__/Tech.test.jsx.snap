// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tech component renders Tech 1`] = `
<section
  className="Tech"
  name="Tech"
>
  <PageHeading
    content="Tech Stack"
  />
  <section
    className="techList"
  >
    <dl
      className="definitionList"
    >
      <TechItem
        definitions={
          Array [
            "JavaScript",
            "HTML",
            "CSS",
          ]
        }
        key="Languages"
        title="Languages"
      />
      <TechItem
        definitions={
          Array [
            "React",
            "React Native",
            "Redux",
            "PostCSS",
          ]
        }
        key="Front End"
        title="Front End"
      />
      <TechItem
        definitions={
          Array [
            "Node",
            "Express",
          ]
        }
        key="Back End"
        title="Back End"
      />
      <TechItem
        definitions={
          Array [
            "MongoDB",
            "Mongoose",
            "PostresSQL",
          ]
        }
        key="Database"
        title="Database"
      />
      <TechItem
        definitions={
          Array [
            "Jest",
            "Enzyme",
            "QUnit",
          ]
        }
        key="Testing Suites"
        title="Testing Suites"
      />
      <TechItem
        definitions={
          Array [
            "Travis CI",
            "Heroku",
            "Postman",
            "Compass",
          ]
        }
        key="Dev Tools"
        title="Dev Tools"
      />
    </dl>
  </section>
  <Arrow
    element="Projects"
  />
</section>
`;
